### PR TITLE
Donea and Huerta book 2D cartesian benchmark

### PR DESCRIPTION
**Pull request by cedrict**
These are the files for a new benchmark: 2D, cartesian, manufactured solution which resembles a convection cell in a unit square, taken from the Donea and Huerta FEM book.

### CODE REVIEW

**Comment by bangerth** [comment]
I'll review this.

File: benchmarks/doneahuerta/doneahuerta.cc [hunk 1]
```diff
@@ -0,0 +1,375 @@
+#include <aspect/simulator.h>
```
Comment (bangerth) [inline benchmarks/doneahuerta/doneahuerta.cc]: can you copy the usual copyright header here?

File: benchmarks/doneahuerta/doneahuerta.cc [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    }
+
+
+
+
+
+
+    /**
+     *gravity model for the DoneaHuerta benchmark
+    */
```
Comment (bangerth) [inline benchmarks/doneahuerta/doneahuerta.cc]: Add a space before 'gravity', capitalize it, and make indentation consistent

File: benchmarks/doneahuerta/doneahuerta.cc [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+      g[1]= ( (8.-48.*y+48.*y*y)*x*x*x + (-12.+72.*y-72*y*y)*x*x
+              +  (4.-24.*y+48.*y*y-48.*y*y*y+24.*y*y*y*y)*x - 12.*y*y + 24.*y*y*y -12.*y*y*y*y);
+      return g;
+    }
+
+    /**
+      * A postprocessor that evaluates the accuracy of the solution.
+      *
+      * The implementation of error evaluators that correspond to the
+      * benchmarks defined in the paper Duretz et al. reference above.
```
Comment (bangerth) [inline benchmarks/doneahuerta/doneahuerta.cc]: update the reference

File: doc/manual/manual.tex [hunk 4]
```diff
@@ -9937,6 +9937,67 @@ \subsubsection{Benchmarks for operator splitting}
...
+This benchmark is taken from Donea and Huerta's book \cite{DH03book}.
+The domain is a unit square and the viscosity and density are set to 1. 
+The components of the gravity vector $\mathbf g$ are prescribed as 
+\begin{eqnarray}
+g_x &=& (12 - 24y) x^4 + (-24 + 48y) x^3 + (-48y + 72y^2 - 48 y^3 + 12) x^2 \nonumber\\
+    && + (-2 + 24y -72y^2+48y^3)x + 1-4y + 12y^2-8y^3 \nonumber\\ 
+g_y &=& (8 - 48y + 48 y^2) x^3 + (-12 + 72y - 72y^2) x^2  \nonumber\\
+    && + (4 - 24y + 48y^2 - 48y^3 + 24y^4) x - 12y^2 + 24y^3 - 12y^4. 
+\end{eqnarray}
+The exact solution is then
```
Comment (bangerth) [inline doc/manual/manual.tex]: maybe say "The exact solution can then be chosen as follows, if one prescribes Dirichlet boundary values for the velocity using the same formula:"

File: doc/manual/manual.tex [hunk 5]
```diff
@@ -9937,6 +9937,67 @@ \subsubsection{Benchmarks for operator splitting}
...
+  \subfigure[]{
+    \includegraphics[width=0.31\textwidth]{cookbooks/benchmarks/doneahuerta/grav.png}}%
+  ~ 
+  \subfigure[]{
+    \includegraphics[width=0.31\textwidth]{cookbooks/benchmarks/doneahuerta/press.png}}%
+  ~
+  \subfigure[]{
+    \includegraphics[width=0.31\textwidth]{cookbooks/benchmarks/doneahuerta/vel.png}}
+  \caption{Donea \& Huerta benchmark: Results for the 2D polynomial Stokes benchmark, 
+obtained with a resolution of $32\times 32$ elements. a) gravity field, b) pressure field,
```
Comment (bangerth) [inline doc/manual/manual.tex]: gravity -> Gravity

File: tests/doneahuerta_3/screen-output.tmp [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+--     . running with 1 MPI process
+-- How to cite ASPECT: https://aspect.geodynamics.org/cite.html
+-----------------------------------------------------------------------------
+
+Loading shared library <./libdoneahuerta.so>
+ERROR: Uncaught exception in MPI_InitFinalize on proc 0. Skipping MPI_Finalize() to avoid a deadlock.
+
+
+----------------------------------------------------
+Exception 'ExcMessage (std::string("Could not successfully load shared library <") + shared_libs_list[i] + ">. The operating system reports " + "that the error is this: <" + dlerror() + ">.")' on rank 0 on processing:
```
Comment (bangerth) [inline tests/doneahuerta_3/screen-output.tmp]: This isn't quite right -- you'll need to add `doneahuerta.cc` that simply does a `#include "..."` to the corresponding file in the benchmarks folder.

File: benchmarks/doneahuerta/doneahuerta.cc (hunk 1 above)
Comment (cedrict) [reply benchmarks/doneahuerta/doneahuerta.cc]: done

File: benchmarks/doneahuerta/doneahuerta.cc (hunk 2 above)
Comment (cedrict) [reply benchmarks/doneahuerta/doneahuerta.cc]: done

File: benchmarks/doneahuerta/doneahuerta.cc (hunk 3 above)
Comment (cedrict) [reply benchmarks/doneahuerta/doneahuerta.cc]: done

File: doc/manual/manual.tex (hunk 4 above)
Comment (cedrict) [reply doc/manual/manual.tex]: agreed, done.

File: doc/manual/manual.tex (hunk 5 above)
Comment (cedrict) [reply doc/manual/manual.tex]: done.

File: tests/doneahuerta_3/screen-output.tmp (hunk 6 above)
Comment (cedrict) [reply tests/doneahuerta_3/screen-output.tmp]: done

File: doc/manual/manual.tex (hunk 4 above)
Comment (bangerth) [reply doc/manual/manual.tex]: Nope, not done :-)

File: tests/doneahuerta_3/screen-output.tmp [hunk 7]
```diff
@@ -0,0 +1,29 @@
+-----------------------------------------------------------------------------
```
Comment (bangerth) [inline tests/doneahuerta_3/screen-output.tmp]: you can remove this file now

**Comment by bangerth** [comment]
@cedrict -- can you reduce this to a smaller, individually self-contained set of commits? Let me know when you're done and we'll merge.

**Pull request merged**